File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/observer/Observer.java
package jv.com.tencent.cain.design_pattern.observer;

/**
 * @author : jiangyu
 * @date : 2020/3/3
 * @desc : 观察者设计模式中的观察者
 */
public interface Observer {
    /**
     * 更新状态
     * @param subject
     * @param data
     */
    void update(Subject subject, Object data);
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/observer/Observer.kt
package kt.com.tencent.cain.desigin_pattern.observer

/**
 * @author : jiangyu
 * @date   : 2020/3/3
 * @desc   : xxx
 */
interface Observer {
    /**
     * 更新状态
     */
    fun update(subject: Subject, data: Any)
}